pyomo/contrib/solver/base.py

```diff
@@ -348,9 +348,20 @@ class LegacySolverWrapper:
     interface. Necessary for backwards compatibility.
     """
 
-    def __init__(self, solver_io=None, **kwargs):
-        if solver_io is not None:
+    def __init__(self, **kwargs):
+        if 'solver_io' in kwargs:
             raise NotImplementedError('Still working on this')
+        # There is no reason for a user to be trying to mix both old
+        # and new options. That is silly. So we will yell at them.
+        if 'options' in kwargs and 'solver_options' in kwargs:
+            raise ApplicationError(
+                "Both 'options' and 'solver_options' were requested. "
+                "Please use one or the other, not both."
+            )
+        elif 'options' in kwargs:
+            self.options = kwargs.pop('options')
+        elif 'solver_options' in kwargs:
+            self.solver_options = kwargs.pop('solver_options')
```

Comment:
Is there a reason we need both the `self.options` and `self.solver_options` attributes? Can't we simplify it down to using the same attribute for both keyword options at this point?

Reply:
It was so a user could do both of these things:
```
solver.options['blah'] = whatever
solver.solver_options['foo'] = 'bar'
```
I don't really care but it seemed logical?

Reply:
That said, this is part of what caused the troubles I ran into. With both of these being present, it was difficult to know which one to assign values to and how each would be used.